src/Controller/ViewerController.php

```diff
@@ -11,37 +11,23 @@
 
 namespace Drupal\collabora_online\Controller;
 
+use Drupal\collabora_online\Cool\CoolUtils;
 use Drupal\Core\Controller\ControllerBase;
 use Drupal\Core\Render\RendererInterface;
-use Drupal\collabora_online\Cool\CoolUtils;
-use Drupal\file\Entity\File;
 use Drupal\media\Entity\Media;
-use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\HttpFoundation\Request;
 
 /**
  * Provides route responses for the Collabora module.
  */
 class ViewerController extends ControllerBase {
 
-    private $renderer;
-
     /**
      * The controller constructor.
      */
-    public function __construct(RendererInterface $renderer) {
-        $this->renderer = $renderer;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public static function create(ContainerInterface $container): self {
-        return new self(
-            $container->get('renderer'),
-        );
-    }
+    public function __construct(
+        private readonly RendererInterface $renderer,
+    ) {}
```

Comment:
This change simplifies the class by using constructor property promotion.
It is now a bit unnecessary in the final version of the PR, but it made more sense with prior commits in this PR where the constructor parameters were changed. I suggest to keep it.

Reply:
so true, bad copy paste